src/main/java/events/writer/PubSubEventWriter.java

```diff
@@ -0,0 +1,172 @@
+            logger.info("Publisher created successfully");
+        } catch (IOException e) {
+            logger.error("Error creating pubsub events.publisher. Error: " + e.getMessage());
+        }
+    }
+
+    /**
+     *
+     * @param projectId
+     * @param topicId
+     */
+    public PubSubEventWriter(String projectId,  String topicId) {
+        Publisher.Builder publisherBuilder = null;
+        this.publisher = null;
+
+        TopicName topicName = TopicName.of(projectId, topicId);
+        try {
+            publisherBuilder = Publisher.newBuilder(topicName);
+            this.publisher = publisherBuilder.build();
+            logger.info("Publisher created successfully");
+        } catch (IOException e) {
+            logger.error("Error creating pubsub events.publisher. Error: " + e.getMessage());
+        }
+    }
+
+    /**
+     *
+     * @param saPath
+     * @return
+     * @throws FileNotFoundException
+     */
+    private InputStream getCredentials(String saPath) throws FileNotFoundException {
+        File credentialsFile = new File(saPath);
+        return new FileInputStream(credentialsFile);
+    }
+
+    /**
+     *
+     * @param builder
+     * @param proxyHost
+     * @param proxyPort
+     * @throws NumberFormatException
+     */
+    private void configureChannelProxy(Publisher.Builder builder, String proxyHost, String proxyPort) throws NumberFormatException {
+        SocketAddress proxySocketAddress = new InetSocketAddress(proxyHost, Integer.parseInt(proxyPort));
+        builder.setChannelProvider(
+                InstantiatingGrpcChannelProvider.newBuilder()
+                        .setChannelConfigurator(managedChannelBuilder -> managedChannelBuilder.proxyDetector(new ProxyDetector() {
+                            @Nullable
+                            @Override
+                            public ProxiedSocketAddress proxyFor(SocketAddress socketAddress) {
+                                if (socketAddress == null) return null;
+
+                                return HttpConnectProxiedSocketAddress.newBuilder()
+                                        .setTargetAddress((InetSocketAddress) socketAddress)
+                                        .setProxyAddress(proxySocketAddress)
+                                        .build();
+                            }
+                        }))
+                        .build()
+        );
+    }
+
+    /**
+     *
+     * @param event
+     */
+    public void publishEvent(Event event) {
+        Gson gson = new Gson();
+
+        logger.info("Publishing event");
+        String stringEvent = gson.toJson(event);
+        ByteString data = ByteString.copyFromUtf8(stringEvent);
+        try {
+            PubsubMessage pubsubMessage = PubsubMessage.newBuilder()
+                    .setData(data)
+                    .build();
+            ApiFuture<String> future = this.publisher.publish(pubsubMessage);
```

Comment:
publisher here can be null if there was an exception while initializing it.

Reply:
done